lang/alan/Portfile

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        alantech alan 0.1.7 v
+revision            0
+
+platforms           darwin
+categories          lang
+
+license             Apache-2 \
+                    AGPL-3
+
+homepage            https://alan-lang.org/
+
+description         The Alan Programming Language
+
+long_description    {*}${description}: The alan compiler and runtime can \
+                    parallelize your code without concurrent or asynchronous \
+                    programming (threads, promises, channels, etc) by only \
+                    allowing iteration and recursion that is guaranteed to \
+                    halt (e.g. no while (true) \{\} loops)
+
+checksums           rmd160  f66acbfb8e35272718d58c8c159d28d234ed8140 \
+                    sha256  c84fd2971bd820bcea659eae6400a50358722f46387510a1fe671c0201739280 \
+                    size    292965
+
+maintainers         {gmail.com:herby.gillot @herbygillot} \
+                    openmaintainer
+
+depends_build       port:cargo \
+                    port:nodejs14 \
+                    port:npm6 \
+                    port:python27 \
+                    port:rust \
+                    port:yarn
+
+build.env-append    PATH=${workpath}/bin:$env(PATH)
+build.target
+
+use_configure       no
```

Comment:
So is this downloading a binary or building from source? If it is a binary download, then please build from source instead. You will have to make sure it is using the right compiler manually.

Reply:
It is building from source.

But it's not using the traditional C/C++ `configure/make` workflow.

`alan` consists of Javascript, Typescript and Rust code, so it's building using `npm`, `yarn` and `rust` (+ fetching Rust packages with `cargo`).

Reply:
OK, wow that's unique!

Reply:
Well it's becoming more widespread.

[deno](https://ports.macports.org/port/deno/summary), the intended successor for NodeJS, is also similarly a mix of Javascript, TypeScript & Rust, but that's currently just distributing the pre-built binary as it was too much hassle getting that to build from source.